stream/address_extractor.js

```diff
@@ -29,59 +29,69 @@ var through = require('through2'),
-      var record = new Document( 'osmaddress', type + '-' + item.getType() + '-' + (item.getId() || ++idOrdinal) );
+      // create a new record for street addresses
+      if( hasValidAddress( doc ) ){
+        var type = isNamedPoi ? 'poi-address' : 'address';
 
-      record
-        .setName( 'default', item.address.number + ' ' + item.address.street )
-        .setCentroid( item.getCentroid() );
+        // copy data to new document
+        var record = new Document( 'osmaddress', type + '-' + doc.getType() + '-' + (doc.getId() || ++idOrdinal) );
 
-      // copy address info
-      record.address = item.address;
+        record
+          .setName( 'default', doc.address.number + ' ' + doc.address.street )
+          .setCentroid( doc.getCentroid() );
 
-      // copy admin data
-      if( item.alpha3 ){ record.setAlpha3( item.alpha3 ); }
-      if( item.admin0 ){ record.setAdmin( 'admin0', item.admin0 ); }
-      if( item.admin1 ){ record.setAdmin( 'admin1', item.admin1 ); }
-      if( item.admin1_abbr ){ record.setAdmin( 'admin1_abbr', item.admin1_abbr ); }
-      if( item.admin2 ){ record.setAdmin( 'admin2', item.admin2 ); }
-      if( item.local_admin ){ record.setAdmin( 'local_admin', item.local_admin ); }
-      if( item.locality ){ record.setAdmin( 'locality', item.locality ); }
-      if( item.neighborhood ){ record.setAdmin( 'neighborhood', item.neighborhood ); }
-      
-      // copy meta data (but maintain the id & type assigned above)
-      record._meta = extend( true, {}, item._meta, { id: record.getId(), type: record.getType() } );
+        // copy address info
+        record.address = doc.address;
+
+        // copy admin data
+        if( doc.alpha3 ){ record.setAlpha3( doc.alpha3 ); }
+        if( doc.admin0 ){ record.setAdmin( 'admin0', doc.admin0 ); }
+        if( doc.admin1 ){ record.setAdmin( 'admin1', doc.admin1 ); }
+        if( doc.admin1_abbr ){ record.setAdmin( 'admin1_abbr', doc.admin1_abbr ); }
+        if( doc.admin2 ){ record.setAdmin( 'admin2', doc.admin2 ); }
+        if( doc.local_admin ){ record.setAdmin( 'local_admin', doc.local_admin ); }
+        if( doc.locality ){ record.setAdmin( 'locality', doc.locality ); }
+        if( doc.neighborhood ){ record.setAdmin( 'neighborhood', doc.neighborhood ); }
+
+        // copy meta data (but maintain the id & type assigned above)
+        record._meta = extend( true, {}, doc._meta, { id: record.getId(), type: record.getType() } );
+
+        this.push( record );
+      }
+
+      // forward doc downstream is it's a POI in it's own right
+      // note: this MUST be below the address push()
+      if( isNamedPoi ){
+        this.push( doc );
+      }
 
-      this.push( record );
     }
 
-    // forward item downstream is it's a POI in it's own right
-    // note: this MUST be below the address push()
-    if( isNamedPoi ){
-      this.push( item );
+    catch( e ){
+      console.error( 'address_extractor error' );
+      console.error( e.stack );
```

Comment:
would it be helpful to output the doc here as well? so it's easier to debug issues.